compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/ResolveUtils.kt

```diff
@@ -428,3 +429,8 @@ fun BodyResolveComponents.transformQualifiedAccessUsingSmartcastInfo(qualifiedAc
     }
     return FirExpressionWithSmartcastImpl(qualifiedAccessExpression, intersectedTypeRef, typesFromSmartCast)
 }
+
+fun CallableId.isInvoke() =
+    packageName == KotlinBuiltIns.BUILT_INS_PACKAGE_FQ_NAME
+            && className?.asString()?.startsWith("Function") == true
```

Comment:
How does it work if `className` is e.g. `KFunction*`?

Reply:
the className needs to start with "Function", so "KFunction*" will not match.

Reply:
Yes, but `KFunction*` also have `invoke` call, and (may be, I don't know exactly) something same should be done for it. I think it's the reason for a test of this case.

Reply:
Tried an example `foo(Int::dec)`, KFunction is generated for arguments, not for parameters. I've added two test cases for KFunction into functions/invoke/invoke.kt.